browser/js/product/products.controller.js

```diff
@@ -36,54 +35,66 @@ app.controller('ProductsController', function ($scope, products) {
 		});
 	});
 
-	$scope.addToCart = function (productId){
-		// add 1 of item to cart.
-		// CartFactory.addToCart(productId, 1);	
-	};
-});
 
-app.controller('ProductController', function ($scope, product, ReviewFactory, Session) {
-	$scope.product = product;
-	$scope.item = {quantity:1};
+		
+	// Code for admin functions:
+	$scope.isAdmin = true;
+	// These are to make the category select checkbox/buttons
 
-	$scope.addToCart = function (){
-		if ($scope.item.quantity<1) $scope.item.quantity=1;
-		$scope.item.product = $scope.product._id;
+	$scope.categoryBools = {};
+	$scope.categoriesToIDs = {};
+	CategoryFactory.fetchAll()
+	.then(categories => 
+		{
+			// $scope.newProduct.categories = categories;
+			$scope.allCategories = categories;
+			categories.forEach(function (elem){
+				$scope.categoryBools[elem.name]=true;
+				$scope.categoriesToIDs[elem.name]=elem._id;
+			});
+			
+		});
 
-		console.log("Item", $scope.item);
-		// $scope.item = {quantity: 1, product: "56d6211a41380ba6524bf3e9"}
+	$scope.$watchCollection('categoryBools', function () {
+		$scope.newProduct.categories = [];
+		angular.forEach($scope.categoryBools, function (value, key) {
+			if (value) {
+				$scope.newProduct.categories.push($scope.categoriesToIDs[key]);
+				// need these to be the Id's not the names..
+			}
+		});
+	});
 
-		// CartFactory.addToCart($scope.item.product, $scope.item.quantity);
+	$scope.newProduct = {
+			categories: [],
+			photoUrl: '/imgBoat/boatImg5.jpg'
+			// Still working on uploading images
 	};
 
-	$scope.areThereProductReviews = function () {
-		return $scope.product.reviews.length>0;
+	$scope.createProduct = function () {
+		ProductsFactory.addProduct($scope.newProduct)
+		.then(function(newProduct){
+			console.log("Boat has been created in db", newProduct);
+			$scope.products.unshift(newProduct);
+			// for some reason this wont update the view...
```

Comment:
Did you try a digest right after this? I mean I would have thought the http request would have done that, but maybe?